hazelcast/src/main/java/com/hazelcast/spi/MigrationAwareService.java

```diff
@@ -114,6 +116,8 @@
      * or failure(s) of any of the migration participants; either master or source or destination.
      * <p>
      * Rollback is not expected to fail at this point, all exceptions will be suppressed and logged.
+     * Implementations of this method must be thread safe as this method may be called concurrently
```

Comment:
minor: maybe we can also move this note to top of the javaDoc to gain more attention.